ansible/roles/internal_lb/tasks/main.yml

```diff
@@ -1,12 +1,27 @@
 - name: Install pip
-  apt:
-    name:
-      - python3-setuptools
-      - virtualenv
-      - python3-pip
-    update_cache: yes
-  async: 600
-  poll: 30
+  block:
+    - name: Install python tools
+      apt:
+        name:
+          - python3-setuptools
+          - virtualenv
+          - python3-pip
+        update_cache: yes
+      async: 120
+      poll: 5
+  rescue:
+    # The above failed, so we need to cleanup and try again
+    - name: Fix interrupted dpkg
+      command: dpkg --configure -a
+    - name: Install python tools
+      apt:
+        name:
+          - python3-setuptools
+          - virtualenv
+          - python3-pip
+        update_cache: yes
+      async: 120
+      poll: 5
```

Comment:
Good question. I think we could remove it, but this felt more logical to me.

The test run that I did indicated that even though the first installation failed (due to time out), the packages were actually installed successfully. This can be seen in the ansible output as this (second) attempt to install the packages had status `OK` not `changed`.
However, I still wanted this task here just to make sure. Otherwise it looks like the installation fails, then we clean up `dpkg` and just carry on. :open_mouth: So I see this as a double check basically.

(To avoid code duplication we could extract the task to a separate file and `include` it instead of writing out the complete task twice, but I think that would just have made it harder to understand.)

Reply:
I think I confused myself here, in my mind the structure of this was something like
```
for i in tries:
  try:
    install_python() # The block part
  catch:
    cleanup() # The rescue part
```

And I thought that the poll was retries, but that's very not the case.
So I definitely think this should be like it is.

Maybe if we loose the async part it's less code copying, but I'm okey leaving it like this as well.

Reply:
:+1: Based on the other comment, I guess we keep the async for now, right?